src/App.tsx

```diff
@@ -1,7 +1,57 @@
-import React from 'react';
+import React, {useState, useEffect, useRef} from 'react';
+import {Player, Filter} from 'tone';
+import {appDeps} from './deps';
+import AudioPlayer from './components/AudioPlayer';
+import AudioFilter from './components/AudioFilter';
 
 const App: React.FC = () => {
-  return <div>Sampler</div>;
+  const [isPlayerLoaded, setPlayerLoaded] = useState<boolean>(false);
+  const player = useRef<Player>();
+  const filter = useRef<Filter>();
+  const {
+    isPlaying,
+    onPlay,
+    onStop,
+    onRestart,
+    setAttribute,
+  } = appDeps.playerController({
+    player: player.current,
+  });
+  const {setFilter} = appDeps.filterController({filter: filter.current});
+
+  useEffect(() => {
+    // Init app dependencies
+    filter.current = appDeps.filterFactory(0, 'allpass', -48);
+    player.current = appDeps.playerFactory(
+      'heal-6.wav',
+      () => setPlayerLoaded(true),
+      [filter.current]
+    );
+
+    player.current.set({loop: true});
+  }, []);
+
+  return (
+    <div>
+      {isPlayerLoaded ? (
+        <>
```

Comment:
do we need these empty tags? what do they do?

Reply:
you're not allowed to return more than one element so a common workaround is using "react fragments" - they're written like this: <React.Fragment></React.Fragment>, but can be shortened to just <></> 

https://reactjs.org/docs/fragments.html